snuba/utils/streams/batching.py

```diff
@@ -115,20 +100,11 @@ class BatchingKafkaConsumer:
         [OFFSET_BEGINNING, OFFSET_END, OFFSET_STORED, OFFSET_INVALID]
     )
 
-    # Set of error codes that can be returned by ``consumer.poll`` calls which
-    # are generally able to be recovered from after a series of retries.
-    RECOVERABLE_ERRORS = frozenset(
-        [
-            KafkaError._PARTITION_EOF,
-            KafkaError._TRANSPORT,  # Local: Broker transport failure
-        ]
-    )
-
     def __init__(
         self,
-        consumer: Consumer,
+        consumer: KafkaConsumer,
         topic: str,
-        worker: AbstractBatchWorker,
+        worker: AbstractBatchWorker[KafkaMessage],
```

Comment:
These type constraints will eventually be loosened but right now too many things try to interact with the `KafkaMessage` to be able to do really do that yet. More details on the plan here: https://github.com/getsentry/snuba/pull/508#discussion_r332665441